src/plugins/intel_gna/src/gna_graph_compiler.cpp

```diff
@@ -2588,10 +2622,12 @@ void GNAGraphCompiler::connectOutput(InferenceEngine::CNNLayerPtr layer, void* p
     // Check that layer will be an output
     if (LayerInfo(layer).isOutput() || !nextLayer) {
         mem_region = REGION_OUTPUTS;
-    }
-    if (LayerInfo(layer).isConst()) {
+    } else if (LayerInfo(layer).isConst()) {
         mem_region = REGION_RO;
+    } else if (!should_use_scratch_allocation(layer, gna_config.gnaFlags.sw_fp32)) {
```

Comment:
In future it would be nice to convert the whole gna_config class to a singleton.